src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -639,38 +639,81 @@ class spell_dk_death_gate : public SpellScriptLoader
         }
 };
 
-// 49560 - Death Grip
-class spell_dk_death_grip : public SpellScriptLoader
+// 49576 - Death Grip Initial
+class spell_dk_death_grip_initial : public SpellScriptLoader
 {
-    public:
-        spell_dk_death_grip() : SpellScriptLoader("spell_dk_death_grip") { }
+public:
+    spell_dk_death_grip_initial() : SpellScriptLoader("spell_dk_death_grip_initial") { }
 
-        class spell_dk_death_grip_SpellScript : public SpellScript
+    class spell_dk_death_grip_initial_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_dk_death_grip_initial_SpellScript);
+
+        SpellCastResult CheckCast()
         {
-            PrepareSpellScript(spell_dk_death_grip_SpellScript);
+            Unit* caster = GetCaster();
+            // Death Grip should not be castable while jumping/falling
+            if (caster->HasUnitState(UNIT_STATE_JUMPING) || caster->HasUnitMovementFlag(MOVEMENTFLAG_FALLING))
+                return SPELL_FAILED_MOVING;
 
-            void HandleDummy(SpellEffIndex /*effIndex*/)
-            {
-                int32 damage = GetEffectValue();
-                Position const* pos = GetExplTargetDest();
-                if (Unit* target = GetHitUnit())
-                {
-                    if (!target->HasAuraType(SPELL_AURA_DEFLECT_SPELLS)) // Deterrence
-                        target->CastSpell(pos->GetPositionX(), pos->GetPositionY(), pos->GetPositionZ(), damage, true);
-                }
-            }
+            // Patch 3.3.3 (2010-03-23): Minimum range has been changed to 8 yards in PvP.
+            Unit* target = GetExplTargetUnit();
+            if (target && target->GetTypeId() == TYPEID_PLAYER)
+                if (caster->GetDistance(target) < 8.f)
```

Comment:
I rather not touch this since I didn't wrote it and I don't know the details of implementation for these methods and it has nothing to do with the issue I'm trying to solve.